### DISCUSSION TITLE
phoebe 2.3

### DISCUSSION THREAD

**Discussion by helenkurd** [phoebe-project/phoebe2#541] [Questions]
hello for entering data to python3 should i write parameters of flux,and phase parameter ? or all parameters in b.add_dataset('lc')
print(b.get_dataset(kind='lc', check_visible=False)) and writing Parameter Set: 30 parameters
times@lc01@dataset: [] d
fluxes@lc01@dataset: [] W / m2
passband@lc01@dataset: Johnson:V
intens_weighting@lc01@dataset: energy
compute_times@lc01@dataset: [] d
C compute_phases@lc01@dataset: []
phases_t0@lc01@dataset: t0_supconj
mask_enabled@lc01@dataset: True
mask_phases@lc01@dataset: []
solver_times@lc01@dataset: auto
sigmas@lc01@dataset: [] W / m2
sigmas_lnf@lc01@dataset: -inf
pblum_mode@lc01@dataset: component-coupled
pblum_component@lc01@dataset: primary
pblum_dataset@lc01@dataset:
pbflux@lc01@dataset: 1.0 W / m2
l3_mode@lc01@dataset: flux
l3@lc01@dataset: 0.0 W / m2
l3_frac@lc01@dataset: 0.0
exptime@lc01@dataset: 0.0 s
ld_mode@primary@lc01@dataset: interp
ld_mode@secondary@lc01@dataset: interp
ld_func@primary@lc01@dataset: logarithmic
ld_func@secondary@lc01@dataset: logarithmic
ld_coeffs_source@primary@lc...: auto
ld_coeffs_source@secondary@...: auto
ld_coeffs@primary@lc01@dataset: [0.5 0.5]
ld_coeffs@secondary@lc01@da...: [0.5 0.5]
pblum@primary@lc01@dataset: 12.566370614359172 W
pblum@secondary@lc01@dataset: 12.566370614359172 W

**Comment by kecnry** [comment]
All of those parameters are optional - you can either pass them when calling `add_dataset` or change the default values later. If you want to attach an observational light curve (for fitting or plotting, rather than just creating a synthetic light curve), then you'll want to set `times`, `fluxes`, and `sigmas` (all with arrays of the same length).

**Comment by helenkurd** [comment]
hello sir ,sir      <ParameterSet: 42 parameters | contexts: compute,
figure, dataset, constraint>    sir whatis mean of parameter set? and
,sir what is mean of 'lc01'?

**Reply by kecnry** [reply]
ParameterSet is a python object containing a filtered list of parameters.  Its introduced in the [first tutorial on general concepts](http://phoebe-project.org/docs/latest/tutorials/general_concepts).

'lc01' is the _default_ label given to the light curve dataset when calling `b.add_dataset` but not providing a string for `dataset`.  If you pass `dataset='mylc'` for example, then it would get that label instead.  You then refer to these various labels when filtering and setting parameters.

**Comment by helenkurd** [comment]
Thanks very much sir for your help

**Comment by helenkurd** [comment]
hellow sir ,sir are this numbers in run compute are constant?or changed if
changed how can i change it?

b.run_compute(irrad_method='none')

100%|██████████| 205/205 [00:00<00:00, 256.51it/s]

**Reply by kecnry** [reply]
What you are seeing there is just the (completed) progress bar for the computation of the forward model (there were 205 iterations/"times" to be computed, and it took less than a second to complete, at the rate of 256.51 iterations per second).  The call to [run_compute](http://phoebe-project.org/docs/latest/api/phoebe.frontend.bundle.Bundle.run_compute) returns the resulting model as a ParameterSet, which you can also access by [filtering on the model context in the bundle](http://phoebe-project.org/docs/latest/tutorials/compute) and can be [plotted directly from PHOEBE](http://phoebe-project.org/docs/latest/tutorials/plotting).

**Comment by helenkurd** [comment]
hello sir for estimating new value of parameters should i write all of below codes:
Generate data
Let's first initialize a bundle and change some of the parameter values. We'll then export the computed models as "observables" to use with the EBAI and lc_geometry estimators.
We'll use blackbody atmospheres with manual limb-darkening just to avoid any out-of-bounds errors if the estimates aren't perfectly physical. Normally we would check these values to make sure they're reasonable before adopting and would definitely want to revert to physical atmospheres and appropriate limb-darkening when optimizing.
b = phoebe.default_binary()
# set parameter values
b.set_value('q', value = 0.6)
b.set_value('incl', component='binary', value = 84.5)
b.set_value('ecc', 0.2)
b.set_value('per0', 63.7)
b.set_value('requiv', component='primary', value=1.)
b.set_value('requiv', component='secondary', value=0.6)
b.set_value('teff', component='secondary', value=5500.)

# add an lc dataset
b.add_dataset('lc', compute_phases=phoebe.linspace(0,1,101))
b.set_value_all('ld_mode', 'manual')
b.set_value_all('ld_mode_bol', 'manual')
b.set_value_all('atm', 'blackbody')

#compute the model
b.run_compute(irrad_method='none')

# extract the arrays from the model that we'll use as observables in the next step
times = b.get_value('times', context='model', dataset='lc01')
# here we're adding noise to the fluxes as well to make the fake data more "realistic"
np.random.seed(0) # to ensure reproducibility with added noise
fluxes = b.get_value('fluxes', context='model', dataset='lc01') + np.random.normal(size=times.shape) * 0.02
sigmas_lc = np.ones_like(times) * 0.04
100%|██████████| 101/101 [00:08<00:00, 11.51it/s]

**Reply by kecnry** [reply]
This seems to just be a copy or modification of the [LC estimators tutorial](http://phoebe-project.org/docs/2.3/tutorials/LC_estimators) which is just a (very) simplified example, using synthetic data, that shows the options for these particular "solvers".  You'll need to significantly modify the script to the needs of your system and science use-cases and likely will need to make use of other solvers as well.

The general workflow, as described in the [Solvers tutorial](http://phoebe-project.org/docs/2.3/tutorials/solver) is to attach your observations, add a set of solver options and modify them as necessary, and call `run_solver` (which then exposes a "solution").  If you're happy with the proposed solution, you can apply them back to your system with `adopt_solution`.

It might be easiest to try to run these notebooks on the synthetic cases shown, understand what each step does, and try modifying some options to see how it reacts.  Hopefully this will make applying the general concepts back to your system more approachable.

**Comment by helenkurd** [comment]
thanks very much sir kecnry butsir for esimating new value of parameters can i use just this below part :
Generate data
Let's first initialize a bundle and change some of the parameter values. We'll then export the computed models as "observables" to use with the EBAI and lc_geometry estimators.

We'll use blackbody atmospheres with manual limb-darkening just to avoid any out-of-bounds errors if the estimates aren't perfectly physical. Normally we would check these values to make sure they're reasonable before adopting and would definitely want to revert to physical atmospheres and appropriate limb-darkening when optimizing.

b = phoebe.default_binary()
# set parameter values
b.set_value('q', value = 0.6)
b.set_value('incl', component='binary', value = 84.5)
b.set_value('ecc', 0.2)
b.set_value('per0', 63.7)
b.set_value('requiv', component='primary', value=1.)
b.set_value('requiv', component='secondary', value=0.6)
b.set_value('teff', component='secondary', value=5500.)

# add an lc dataset
b.add_dataset('lc', compute_phases=phoebe.linspace(0,1,101))
b.set_value_all('ld_mode', 'manual')
b.set_value_all('ld_mode_bol', 'manual')
b.set_value_all('atm', 'blackbody')

#compute the model
b.run_compute(irrad_method='none')

**Comment by helenkurd** [comment]
hello sir how are you sir to plot spots of my stars which are W UM a contact binary stars in spot tutorials i saw this numbers below of time=[0,0.25,0.5,0.75,1.0] is this numbers constant for all contact binary stars ? can i use same steps of this below codes for my contact W UMa binary stars To see the spot, add a mesh dataset and plot it. Thank you very much for your suggestion 

To see the spot, add a mesh dataset and plot it.

b.add_dataset('mesh', times=[0,0.25,0.5,0.75,1.0], columns=['teffs'])
Wed, 24 Feb 2021 10:36 BUNDLE WARNING mesh dataset uses 'compute_times' instead of 'times', applying value sent as 'times' to 'compute_times'.
<ParameterSet: 85 parameters | contexts: dataset, constraint, compute, figure>
b.run_compute()
100%|██████████| 5/5 [00:00<00:00, 12.76it/s]
<ParameterSet: 53 parameters | components: primary, secondary>
afig, mplfig = b.filter(component='primary', time=0.75).plot(fc='teffs', show=True)

**Reply by kecnry** [reply]
The times list/array are the times at which you want the mesh computed and exposed (and then available for plotting).  Optionally, you can pass [compute_phases](http://phoebe-project.org/docs/latest/tutorials/compute_times_phases) instead if that is more convenient.

**Comment by helenkurd** [comment]
Thanks in advance for your reply
